rdsbroker/broker.go

```diff
@@ -387,20 +391,79 @@ func (b *RDSBroker) LastOperation(instanceID string) (brokerapi.LastOperationRes
 	return lastOperationResponse, nil
 }
 
+func (b *RDSBroker) CheckAndRotateCredentials() error {
+	dbInstanceDetailsList, err := b.dbInstance.DescribeByTag("Broker Name", b.brokerName)
+	if err != nil {
+		b.logger.Error("Could not retrieve the list of DB instances", err)
+		return err
+	}
+
+	for _, dbInstanceDetails := range dbInstanceDetailsList {
+		dbAddress := dbInstanceDetails.Address
+		dbPort := dbInstanceDetails.Port
+		masterUsername := dbInstanceDetails.MasterUsername
+		serviceInstanceID := b.dbInstanceIdentifierToServiceInstanceID(dbInstanceDetails.Identifier)
+
+		var dbName string
+		if dbInstanceDetails.DBName != "" {
+			dbName = dbInstanceDetails.DBName
+		} else {
+			dbName = b.dbName(dbInstanceDetails.Identifier)
+		}
+
+		sqlEngine, err := b.sqlProvider.GetSQLEngine(dbInstanceDetails.Engine)
+		if err != nil {
+			b.logger.Error("Could not get the SQL Engine", err)
+			return err
+		}
+
+		err = sqlEngine.Open(dbAddress, dbPort, dbName, masterUsername, b.masterPassword(serviceInstanceID))
+		if err != nil {
+			b.logger.Error("Error opening connection to the database", err)
+			return err
+		} else {
+			defer sqlEngine.Close()
```

Comment:
Minor: can you put the `defer` outside the `if` condition? We're going to continue at this point anyway, so it doesn't need to be conditional and it's a bit easier to read.